### DISCUSSION TITLE
Change shadow color

### DISCUSSION THREAD

**Discussion by dvdcarlomagno** [google/model-viewer#4154] [Q&A]
Hi there! I'm placing my 3d model on a dark background and I would need to have some contrast by using a white-ish shadow of my render. Is there any API that allows to change the color of the shadow? Thank you in advance.

**Comment by elalish** [comment]
No, we don't. Would a white shadow look good? Seems not very realistic. Maybe you could share some screenshots of good examples?

**Reply by dvdcarlomagno** [reply]
I know it might sound strange to have a light shadow and it might sound contradictory as well... but here's my specific case.
<img width="1386" alt="image" src="https://user-images.githubusercontent.com/36045489/224309059-08e5e9aa-f940-4cf8-be00-88399b74685e.png">

**Reply by elalish** [reply]
Have you tried photo-shopping in a white shadow? I'm dubious it would improve the look. Especially considering the dark vehicle, I'd recommend a lighter background anyway simply so you have some contrast.

**Reply by dvdcarlomagno** [reply]
Yes, I did try photo-shopping it. I know that the car color and the background don't help but this is the client's palette 🤷🏼‍♂️
Well, thank you for your answer tho. I really appreciate it. I guess I'll try to figure out another solution.

**Comment by shi11** [comment]
Just wanted to respond to this after discovering this from a search. Ultimately my solution was to change the background using to a radial-gradient(#222222, #000000) - giving a little grey to the center of the object allowing the shadow to show up nicely.

**Comment by fiko70** [comment]
Hi @elalish
to your question actually, I do have a real use case ... **in AR, when the shadow should be a bloom effect projected on the floor.**
Please see screenshot with a well known example from the lib ... should the _shadow_ be white or yellow, it would be much more convincing for an emissive object.
![shadow that should be light](https://github.com/google/model-viewer/assets/168015903/7e62617d-bdca-4913-8265-8bf26ea076f5)

I am trying to do a small project with many lights and bulbs in AR, so if anyone can help with either:

- how to change this "shadow" color ' (parameter, or hardcoded change in .js ?) (just change a color code somewhere ? or "+" instead of "-" to lighten instead of darken ? sorry, not a coder at all, but principle looks very simple)
- or to achieve the same effect by disabling the shadow & using something else (a flat round white gradient texture always stays dynamically on the ground plane, that scales with the model, could do the trick)

But one thing for sure, is that this effect or shadow on the surrounding reality is what make the object included, not just pasted on the screen. So a must have.
Help super appreciated,
F

**Reply by elalish** [reply]
Huh, yeah it would be cool for a punctual light in a glTF to light up the AR scene, but that sounds hard for a host of reasons. If you figure out a good way, PRs are welcome. A custom shadow color sounds way too hacky for what you really want to achieve.

**Comment by fiko70** [comment]
Hi Elalish,
"_A custom shadow color sounds way too hacky for what you really want to achieve_"  ... not at all. The objective of the shadow, or glow is to show that the object is lying on the floor/table/reality and not floating in mid air.

Please tell me/us in the .js code where to hardcode that different color or brightness inversion. I am not asking some nice parameter passing and complete lib change.

**Reply by elalish** [reply]
Just search for "shadow" in our code base and you should find it. You may need to change the shader.
